components/TxForm.tsx

```diff
@@ -190,9 +190,25 @@ export const TxForm = () => {
         </InputGroup>
         <Flex justify="flex-end"></Flex>
       </FormControl>
-      <Button size="lg" mt="1rem" width="100%" onClick={doSubmit}>
-        Shield
-      </Button>
+      {needsApproval ? (
+        <Button
+          size="lg"
+          mt="1rem"
+          width="100%"
+          onClick={async () => {
+            if (!doErc20Approval) {
+              throw "not prepared";
```

Comment:
Instead of throwing can we wait for the preparation to finish?

Reply:
Yes, we can should probably pop a notification instead